### PR TITLE
reef: rgw/multisite: don't retain RGW_ATTR_OBJ_REPLICATION_TRACE attr on copy_object

### PR DESCRIPTION
**Pull request by smanjara**
backport tracker: https://tracker.ceph.com/issues/67026

---

backport of https://github.com/ceph/ceph/pull/58519
parent tracker: https://tracker.ceph.com/issues/66983

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smanjara** [comment]
jenkins test make check

File: src/rgw/driver/rados/rgw_rados.cc [hunk 1]
```diff
@@ -4418,6 +4418,10 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
     src_attrs.erase(RGW_ATTR_OLH_VER);
   }
 
+  src_attrs.erase(RGW_ATTR_OBJ_REPLICATION_TRACE);
+  src_attrs.erase(RGW_ATTR_OBJ_REPLICATION_TIMESTAMP);
```
Comment (cbodley) [inline src/rgw/driver/rados/rgw_rados.cc:4422]: RGW_ATTR_OBJ_REPLICATION_TIMESTAMP doesn't exist on reef:
```
src/rgw/driver/rados/rgw_rados.cc:4422:19: error: ‘RGW_ATTR_OBJ_REPLICATION_TIMESTAMP’ was not declared in this scope; did you mean ‘RGW_ATTR_OBJ_REPLICATION_TRACE’?
 4422 |   src_attrs.erase(RGW_ATTR_OBJ_REPLICATION_TIMESTAMP);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment (smanjara) [reply src/rgw/driver/rados/rgw_rados.cc:4422]: ha thanks

**Comment by smanjara** [comment]
@cbodley can you approve this please?

**Review by cbodley — APPROVED** [review]

**Pull request merged**